go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -330,7 +330,14 @@ type builtinConvertTz struct {
 
 var _ IR = (*builtinConvertTz)(nil)
 
-func convertTz(dt datetime.DateTime, from, to *time.Location) (datetime.DateTime, bool) {
+func convertTz(dt datetime.DateTime, from, to *time.Location, now time.Time) (datetime.DateTime, bool) {
+	t := dt.ToStdTime(now)
+	lowerBoundTz := time.Date(1970, 01, 01, 0, 0, 0, 0, time.UTC)
+	upperBoundTz := time.Date(3001, 1, 19, 3, 14, 7, 99999, time.UTC)
```

Comment:
Is this based on https://dev.mysql.com/doc/dev/mysql-server/latest/my__time_8h.html#a67b86a393410bf1e9ab66ba46764c33d? It seems like it's slightly off from that value.

Reply:
We already have a `const maxUnixtime = 32536771200` which represents this value anyway. So I think we should use that. We also should add a date this far in the future then to the tests I think to ensure the max value is also tested (and not just pre-unix epoch times).

Reply:
https://dev.mysql.com/doc/refman/8.3/en/date-and-time-functions.html#function_unix-timestamp

Says that for UNIX_TIMESTAMP is range is 1970-01-01 00:00:01.000000' UTC to '3001-01-19 03:14:07.999999' UTC (corresponding to 32536771199.999999 seconds).

And, 
https://dev.mysql.com/doc/refman/8.3/en/date-and-time-functions.html#function_convert-tz

Says that the range for CONVERT_TZ is '1970-01-01 00:00:01' UTC to '3001-01-18 23:59:59.999999' UTC.

I think I should update the range for `CONVERT_TZ`?

Reply:
> [I checked, I think it should be `3001-01-18 23:59:59.999999` UTC, no idea why it says `3001-01-19 03:14:07.999999`]

Yeah, the docs are not always actually correct, the code speaks the truth usually :smile:.